lib/services/vote_service.dart

```diff
@@ -13,30 +15,49 @@ class VoteService {
   VoteService._();
 
   final ItemApi _itemApi = new ItemApi();
+  final CommentApi _commentApi = new CommentApi();
   CachedVoteRepository _voteRepository = CachedVoteRepository.instance;
 
-  Future voteItem(Item item, Vote vote) async {
+  Future<String> get nonce async {
     final prefs = await SharedPreferences.getInstance();
-    if (prefs.containsKey(SharedPrefKeys.MeToken)) {
-      final urlEncodedMeCookie = prefs.getString(SharedPrefKeys.MeToken);
-      final meToken = MeCookie.fromUrlEncodedJson(urlEncodedMeCookie);
-      final nonce = meToken.id.substring(0, 16);
-
-      _itemApi.vote(item.id, vote, nonce);
-      _voteRepository.saveVote(
-        itemId: item.id,
-        itemType: ItemType.item,
-        vote: vote,
-      );
-    }
+    if (!prefs.containsKey(SharedPrefKeys.MeToken)) return null;
+    final urlEncodedMeCookie = prefs.getString(SharedPrefKeys.MeToken);
+    final meToken = MeCookie.fromUrlEncodedJson(urlEncodedMeCookie);
+    return meToken.id.substring(0, 16);
+  }
+
+  Future voteItem(Item item, Vote vote) async {
```

Comment:
voteItem and voteComment are nearly identical. Make a common method that they both use an only pass the ItemType